de.metas.vertical.pharma.msv3.server-peer-metasfresh/src/main/java/de/metas/vertical/pharma/msv3/server/peer/metasfresh/interceptor/C_BPartner_Product.java

```diff
@@ -0,0 +1,129 @@
+package de.metas.vertical.pharma.msv3.server.peer.metasfresh.interceptor;
+
+import org.adempiere.ad.modelvalidator.annotations.Interceptor;
+import org.adempiere.ad.modelvalidator.annotations.ModelChange;
+import org.adempiere.ad.trx.api.ITrxListenerManager.TrxEventTiming;
+import org.adempiere.ad.trx.api.ITrxManager;
+import org.adempiere.exceptions.FillMandatoryException;
+import org.adempiere.model.InterfaceWrapperHelper;
+import org.adempiere.util.Check;
+import org.adempiere.util.Services;
+import org.compiere.Adempiere;
+import org.compiere.model.I_C_BPartner_Product;
+import org.compiere.model.ModelValidator;
+import org.springframework.stereotype.Component;
+
+import de.metas.vertical.pharma.msv3.server.peer.metasfresh.services.MSV3StockAvailabilityService;
+import lombok.NonNull;
+
+/*
+ * #%L
+ * metasfresh-pharma.msv3.server-peer-metasfresh
+ * %%
+ * Copyright (C) 2018 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+@Interceptor(I_C_BPartner_Product.class)
+@Component("de.metas.vertical.pharma.msv3.server.peer.metasfresh.interceptor.C_BPartner_Product")
+public class C_BPartner_Product
+{
+	private MSV3StockAvailabilityService getStockAvailabilityService()
```

Comment:
Done. I moved all the 3 private methods bellow all the public methods because they are called by more than 1 of them each. Since the methods are not too long and jumping to them is easy and close, I think it's ok this way. Also it repects the newspaper metaphor